src/__tests__/client.ts

```diff
@@ -1695,6 +1695,7 @@ describe('reconnecting', () => {
     console.warn = warn;
     await testCloseCode(CloseCode.InternalServerError);
     await testCloseCode(CloseCode.InternalClientError);
+    await testCloseCode(CloseCode.BadGateway);
```

Comment:
This does not belong here, the test states "should report some close events immediately and not reconnect". The client _should_ reconnect on a 1014.

```suggestion
```

Reply:
just acknowledging that I read this! it might take me a bit to write a unit test because I'm used to mocking apollo-specific web client servers and am only working on this on weekends 😅

Reply:
If you're busy - I can finish this. The required changes are ultra minimal - basically removing code.